### ISSUE TITLE
Limited Language Support in the `LANGUAGE` Environment (5 languages)

### ISSUE THREAD

**Issue by weeix** [GeoNode/geonode#11633]
## Expected Behavior

The application should ideally support a wider range of languages beyond the currently supported ones, which include Deutsch (de-de), English (en-us), Español (es-es), Français (fr-fr), and Italiano (it-it). Even if a language is not available in MapStore, it should still be supported as the translations available in the [GeoNode repository](https://github.com/GeoNode/geonode/tree/b30312136a1c903d9e4b6ee568f23a56e573dc6c/geonode/locale) can be utilized. This approach is preferable to not having any support at all.

## Actual Behavior

At present, only the aforementioned five languages are supported. If any other language is specified, it is silently ignored and removed. The code that filters the LANGUAGE environment can be found [here](https://github.com/GeoNode/geonode/blob/b30312136a1c903d9e4b6ee568f23a56e573dc6c/geonode/settings.py#L1564-L1578).

## Steps to Reproduce the Problem

  1. Clone the GeoNode repository with the command: `git clone https://github.com/GeoNode/geonode.git -b 4.1.3`
  2. Navigate to the cloned directory: `cd geonode`
  3. Modify the `.env` file. Change:
     
      ```
      # LANGUAGE_CODE=pt
      # LANGUAGES=(('en-us','English'),('it-it','Italiano'))
      ```
    
      to:
    
      ```
      LANGUAGE_CODE=en
      LANGUAGES=(('en-us','English'),('it-it','Italiano'),('th-th','ภาษาไทย'))
      ```
    
  4. Build the Docker image with: `docker-compose build --no-cache`
  5. Start the application with: `docker-compose up`

## Specifications

  - GeoNode version: 4.1.3
  - Installation type: vanilla
  - Installation method: Docker
  - Platform: Windows (Rancher Desktop)
  - Additional details:

**Comment by joostvenema** [comment]
Yes, +1 for this issue. More supported languages would be very welcome. It seems there are more translations available at this moment (in Geonode as well as in the geonode-mapstore-client). Can't those just be added to the default list of supported languages?

**Comment by TheHoboCoder** [comment]
Hello, can someone provide status on this issue? It's very confusing, that it seems like some language is supported, there's translations for it, but actually it is not. I think it should be mentioned more clearly in the docs that only these five languages supported at the moment.

**Comment by TheHoboCoder** [comment]
If [here](https://github.com/GeoNode/geonode-mapstore-client/blob/master/geonode_mapstore_client/static/mapstore/) translation is present, you can try override LANGUAGES setting inside container to ignore filtering out. Not tried yet, though. 
However, there's another issue https://github.com/GeoNode/geonode-mapstore-client/issues/500, so high chance it won't work

**Comment by ridoo** [comment]
Perhaps you just can edit your `settings.py` to provide [available languages](https://github.com/GeoNode/geonode/blob/master/geonode/settings.py#L192) as you like:

https://github.com/GeoNode/geonode/blob/5d5a99a6183c9d5fc626072ce68c32fe3f292ffe/geonode/settings.py#L1548-L1553

**Comment by giohappy** [comment]
Closing as part of a repository cleanup: this issue was last updated before 2025-01-01. Not a judgment on the report itself — feel free to reopen or file a fresh issue against `main` if it's still relevant. Thanks for the report.
